### PR TITLE
feat(hatchery): inject env variables into workers from hatchery config

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/service/types.go [hunk 1]
```diff
@@ -46,13 +46,14 @@ type HatcheryCommonConfiguration struct {
...
-		MaxConcurrentProvisioning int    `toml:"maxConcurrentProvisioning" default:"10" comment:"Maximum allowed simultaneous workers provisioning" json:"maxConcurrentProvisioning"`
-		MaxConcurrentRegistering  int    `toml:"maxConcurrentRegistering" default:"2" comment:"Maximum allowed simultaneous workers registering. -1 to disable registering on this hatchery" json:"maxConcurrentRegistering"`
-		RegisterFrequency         int    `toml:"registerFrequency" default:"60" comment:"Check if some worker model have to be registered each n Seconds" json:"registerFrequency"`
-		Region                    string `toml:"region" default:"" comment:"region of this hatchery - optional. With a free text as 'myregion', user can set a prerequisite 'region' with value 'myregion' on CDS Job" json:"region"`
-		IgnoreJobWithNoRegion     bool   `toml:"ignoreJobWithNoRegion" default:"false" comment:"Ignore job without a region prerequisite if ignoreJobWithNoRegion=true"`
+		InjectEnvVars             []string `toml:"injectEnvVars" commented:"true" comment:"Inject env variables in workers" json:"injectEnvVars,omitempty" mapstructure:"injectEnvVars"`
```
Comment (richardlt) [inline engine/service/types.go:49]: json:"-" ?

File: engine/config.go [hunk 2]
```diff
@@ -115,6 +115,7 @@ func configBootstrap(args []string) Configuration {
 			conf.Hatchery.Openstack.HTTP.Port = 8086
 		case sdk.TypeHatchery + ":swarm":
 			conf.Hatchery.Swarm = &swarm.HatcheryConfiguration{}
+			conf.Hatchery.Swarm.Provision.InjectEnvVars = []string{"CDS=1"}
```
Comment (richardlt) [inline engine/config.go:118]: To remove ?

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15994.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/service/types.go (hunk 1 above)
Comment (fsamin) [reply engine/service/types.go:49]: right

File: engine/config.go (hunk 2 above)
Comment (fsamin) [reply engine/config.go:118]: i think i can be an example, but i will remove it

**Review by sguiheux — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**